addons/web/static/src/oxp/todo_item.js

```diff
@@ -0,0 +1,36 @@
+/** @odoo-module **/
+
+import { Component, useState } from "@odoo/owl";
+
+export class TodoItem extends Component {
+    static template = "TodoItem";
+    static props = {
+        todoItem: Object,
+        toggleDone: Function,
+        delete: Function,
+        editMessage: Function,
+    };
+
+    setup() {
+        this.state = useState({
+            isInEdition: false,
+        });
+    }
+
+    delete() {
+        this.props.delete();
```

Comment:
not sure, but I usually inline these functions directly in the template: `props.delete`